tony-core/src/main/java/com/linkedin/tony/util/Utils.java

```diff
@@ -431,6 +431,17 @@ public static void addResources(String[] resources, Map<String, LocalResource> r
   public static void addResource(String path, Map<String, LocalResource> resourcesMap, FileSystem fs) {
     try {
       if (path != null) {
+        // Check the format of the path, if the path is of path::archive, we set resource type as ARCHIVE
+        String[] pathType = path.split("::");
+        if (pathType.length == 2 && pathType[1].equals("archive")) {
+          assert Utils.isArchive(pathType[0]); // Assert it is an arvhive file.
```

Comment:
`Utils.isArchive()` currently only works for local archive files. Can we also support remote archive files, e.g.: zips that are already on HDFS?

A few other minor things:
* I think `Preconditions.assertTrue(...)` is more idiomatic than `assert ...`
* `arvhive` --> `archive`

Reply:
makes sense